cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceHandler.java

```diff
@@ -231,6 +231,9 @@ public void update(HttpRequest request, HttpResponder responder, @PathParam("nam
   private DatasetInstanceConfiguration  getInstanceConfiguration(HttpRequest request) {
     Reader reader = new InputStreamReader(new ChannelBufferInputStream(request.getContent()), Charsets.UTF_8);
     DatasetInstanceConfiguration creationProperties = GSON.fromJson(reader, DatasetInstanceConfiguration.class);
+    if (creationProperties.getProperties() == null) {
```

Comment:
Instead of having a default value for properties, could we just handle null properties somewhere in `DatasetInstanceHandler`? I feel like that would be the most simple way.